internal/jet/literal_expression.go

```diff
@@ -160,6 +161,66 @@ func Decimal(value string) FloatExpression {
 	return &floatLiteral
 }
 
+// ---------------------------------------------------//
+
+type boolArrayLiteral struct {
+	arrayInterfaceImpl[BoolExpression]
+	literalExpressionImpl
+}
+
+func BoolArray(values []bool) ArrayExpression[BoolExpression] {
+	l := boolArrayLiteral{}
+	l.literalExpressionImpl = *literal(pq.BoolArray(values))
+	l.arrayInterfaceImpl.parent = &l
+
+	return &l
+}
+
+type integerArrayLiteral struct {
+	arrayInterfaceImpl[IntegerExpression]
+	literalExpressionImpl
+}
+
+func Int64Array(values []int64) ArrayExpression[IntegerExpression] {
+	l := integerArrayLiteral{}
+	l.literalExpressionImpl = *literal(pq.Int64Array(values))
+	l.arrayInterfaceImpl.parent = &l
+	return &l
+}
+
+func Int32Array(values []int32) ArrayExpression[IntegerExpression] {
+	l := integerArrayLiteral{}
+	l.literalExpressionImpl = *literal(pq.Int32Array(values))
+	l.arrayInterfaceImpl.parent = &l
+	return &l
+}
+
+type stringArrayLiteral struct {
+	arrayInterfaceImpl[StringExpression]
+	literalExpressionImpl
+}
+
+func StringArray(values []string) ArrayExpression[StringExpression] {
+	l := stringArrayLiteral{}
+	l.literalExpressionImpl = *literal(pq.StringArray(values))
+	l.arrayInterfaceImpl.parent = &l
+
+	return &l
+}
+
+type unsafeArrayLiteral[E Expression] struct {
+	arrayInterfaceImpl[E]
+	literalExpressionImpl
+}
+
+func UnsafeArray[E LiteralExpression](values []interface{}) ArrayExpression[E] {
```

Comment:
Why is it needed?

Reply:
Idea was to support any type of array, but better not to include it